app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java

```diff
@@ -38,7 +45,39 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         if (mView == null) {
             mView = inflater.inflate(R.layout.fragment_search, container, false);
+
+            if (!Connections.isNetworkAvailable(mActivity)) {
+                LinearLayout errorView = (LinearLayout) mView.findViewById(R.id.error_view);
+                TextView errorMessage = (TextView) mView.findViewById(R.id.error_message);
+                ImageView errorIcon = (ImageView) mView.findViewById(R.id.error_icon);
+
+                errorView.setVisibility(View.VISIBLE);
+                errorMessage.setText(mActivity.getString(R.string.error_message_network));
+                errorIcon.setImageResource(R.drawable.ic_cloud_off_black_36dp);
+                return mView;
+            }
+
+            SearchFetchTask searchFetchTask = new SearchFetchTask(mActivity, new AsyncTaskCallbackInterface() {
+                @Override
+                public void onOperationComplete(Bundle bundle) {
+                    if (bundle != null && bundle.getBoolean("valid_return")) {
+                        setupTabs(bundle);
+                        LinearLayout tabLayout = (LinearLayout) mView.findViewById(R.id.search_detail_layout);
+                        tabLayout.setVisibility(View.VISIBLE);
+                    } else {
+                        LinearLayout errorView = (LinearLayout) mView.findViewById(R.id.error_view);
+                        TextView errorMessage = (TextView) mView.findViewById(R.id.error_message);
```

Comment:
in this condition, need to set the retry button visibility gone or something like that.